### PR TITLE
[function](round) compute accurate round value by decimal

### PR DESCRIPTION
**Pull request by Gabriel39**
# Proposed changes

before:
image

after:
image

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.81 seconds
 load time: 432 seconds
 storage size: 17123356289 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221208151116_clickbench_pr_60447.html

File: be/src/vec/functions/round.h [hunk 1]
```diff
@@ -0,0 +1,555 @@
...
+
+template <typename T, RoundingMode rounding_mode, ScaleMode scale_mode,
+          TieBreakingMode tie_breaking_mode>
+struct IntegerRoundingComputation {
+    static const size_t data_count = 1;
+
+    static size_t prepare(size_t scale) { return scale; }
+
+    /// Integer overflow is Ok.
+    static ALWAYS_INLINE T computeImpl(T x, T scale) {
```
Comment (HappenLee) [inline be/src/vec/functions/round.h:69]: `compute_impl`

**Review by HappenLee — APPROVED** [review]
LGTM

**Comment by yangzhg** [comment]
round(16.025,2) should be 16.03 not 16.03000

**Pull request merged**